src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c

```diff
@@ -1314,36 +1313,36 @@ static void xclmgmt_extended_probe(struct xclmgmt_dev *lro)
 	    (xocl_subdev_is_vsec(lro) || dev_info->flags & XOCL_DSAFLAG_CUSTOM_DTB))
 		ret = -ENODEV;
 
-	if (!ret) {
-		xocl_thread_start(lro);
-
-		/* Launch the mailbox server. */
-		(void) xocl_peer_listen(lro, xclmgmt_mailbox_srv,
-			(void *)lro);
-
-		lro->ready = true;
-	} else if (ret == -ENODEV) {
+	/* Validate the above two statements */
+	if (ret == -ENODEV) { /* If no device was found attempt to load the device tree*/
 		ret = xclmgmt_load_fdt(lro);
-		if (ret)
+		if (ret) {
+			mgmt_err(lro, "%s\n", "Failed to load FDT");
 			goto fail_all_subdev;
-	} else
+		}
+	} else if (ret) { /* For general failures reset to minimum initalization */
+		mgmt_err(lro, "%s\n", "Firmware ICAP download failed");
 		goto fail_all_subdev;
+	}
+
+	/* Finish initialization. Getting here means nothing failed */
+	xocl_thread_start(lro);
+
+	/* Launch the mailbox server. */
+	(void) xocl_peer_listen(lro, xclmgmt_mailbox_srv, (void *)lro);
 
 	/* Reset PCI link monitor */
 	check_pcie_link_toggle(lro, 1);
 
 	/* Store/cache PCI link width & speed info */
 	store_pcie_link_info(lro);
-
-	/* Notify our peer that we're listening. */
-	xclmgmt_connect_notify(lro, true);
```

Comment:
Why is notify() removed? what if xocl driver is already up and running at this time?

Reply:
That was moved into the end of xclmgmt_probe. I found it to be more readable that the xclmgmt driver notifies xocl it is ready when the probe is complete.

Reply:
It does not seem right. That means all the failure cases will be marked "ready" as well.

Reply:
Moved it back in here